java-squid/src/main/java/org/sonar/java/ast/parser/JavaParser.java

```diff
@@ -39,4 +45,26 @@ private JavaParser() {
       JavaLexer.COMPILATION_UNIT);
   }
 
+  @Override
+  public Tree parse(File file) {
+    return createParentLink((JavaTree) super.parse(file));
+  }
+
+  @Override
+  public Tree parse(String source) {
+    return createParentLink((JavaTree) super.parse(source));
+  }
+
+  private static Tree createParentLink(JavaTree parent) {
+    if (!parent.isLeaf()) {
+      for (Iterator<Tree> iter = parent.childrenIterator(); iter.hasNext();) {
+        JavaTree next = (JavaTree) iter.next();
+        if (next != null) {
+          next.setParent(parent);
```

Comment:
This is a known issue but not relevant to this ticket : https://jira.sonarsource.com/browse/SONARJAVA-1176